pkg/api/api.go

```diff
@@ -0,0 +1,410 @@
+			prevDays:   allPrev.TestPassPercentage,
+		},
+	}
+
+	return &summary
+}
+
+// stats on failure groups
+func failureGroups(failureGroups, failureGroupsPrev []util.JobRunResult, endDay int) *FailureGroups {
+
+	_, _, median, medianPrev, avg, avgPrev := util.ComputeFailureGroupStats(failureGroups, failureGroupsPrev)
+
+	latestDays := fmt.Sprintf("latest%dDays", endDay)
+	prevDays := "prev7Days"
+
+	failureGroupStruct := FailureGroups{
+		JobRunsWithFailureGroup: map[string]int{
+			latestDays: len(failureGroups),
+			prevDays:   len(failureGroupsPrev),
+		},
+		AvgFailureGroupSize: map[string]int{
+			latestDays: avg,
+			prevDays:   avgPrev,
+		},
+		MedianFailureGroupSize: map[string]int{
+			latestDays: median,
+			prevDays:   medianPrev,
+		},
+	}
+	return &failureGroupStruct
+}
+
+func summaryJobsByPlatform(report, reportPrev util.TestReport, endDay, jobTestCount int) []JobSummaryPlatform {
+	jobsByPlatform := util.SummarizeJobsByPlatform(report)
+	jobsByPlatformPrev := util.SummarizeJobsByPlatform(reportPrev)
+
+	latestDays := fmt.Sprintf("latest%dDays", endDay)
+	prevDays := "prev7Days"
+
+	var jobSummariesByPlatform []JobSummaryPlatform
+
+	for _, v := range jobsByPlatform {
+		prev := util.GetPrevPlatform(v.Platform, jobsByPlatformPrev)
+
+		var jobSummaryPlatform JobSummaryPlatform
+
+		if prev != nil {
+			jobSummaryPlatform = JobSummaryPlatform{
+				Platform: v.Platform,
+				PassRates: map[string]PassRate{
+					latestDays: PassRate{
+						Percentage:          v.PassPercentage,
+						ProjectedPercentage: v.PassPercentageWithKnownFailures,
+						Runs:                v.Successes + v.Failures,
+					},
+					prevDays: PassRate{
+						Percentage:          prev.PassPercentage,
+						ProjectedPercentage: prev.PassPercentageWithKnownFailures,
+						Runs:                prev.Successes + prev.Failures,
+					},
+				},
+			}
+		} else {
+			jobSummaryPlatform = JobSummaryPlatform{
+				Platform: v.Platform,
+				PassRates: map[string]PassRate{
+					latestDays: PassRate{
+						Percentage:          v.PassPercentage,
+						ProjectedPercentage: v.PassPercentageWithKnownFailures,
+						Runs:                v.Successes + v.Failures,
+					},
+				},
+			}
+		}
+
+		jobSummariesByPlatform = append(jobSummariesByPlatform, jobSummaryPlatform)
+	}
+	return jobSummariesByPlatform
+}
+
+// top failing tests with a bug
+func summaryTopFailingTestsWithBug(topFailingTestsWithBug []*util.TestResult, resultPrev map[string]util.SortedAggregateTestResult, endDay int) []FailingTestBug {
+
+	latestDays := fmt.Sprintf("latest%dDays", endDay)
+	prevDays := "prev7Days"
+	var topFailingTests []FailingTestBug
+
+	allPrev := resultPrev["all"]
+
+	for _, test := range topFailingTestsWithBug {
+		encodedTestName := url.QueryEscape(regexp.QuoteMeta(test.Name))
+
+		testLink := fmt.Sprintf("https://search.svc.ci.openshift.org/?maxAge=168h&context=1&type=bug%%2Bjunit&name=&maxMatches=5&maxBytes=20971520&groupBy=job&search=%s", encodedTestName)
```

Comment:
make the format string a constant and reuse it in html.go so we don't have two copies of the value potentially getting out of sync